File: Arcade/1-Intro/9-DarkWilderness/knapsackLight.py
# https://codefights.com/arcade/intro/level-9/r9azLYp2BDZPyzaG2/
# Nice solution by virgile_f
def knapsackLight(value1, weight1, value2, weight2, maxW):
    return max( int(weight1<=maxW)*value1, 
                int(weight2<=maxW)*value2, 
                int(weight1+weight2<=maxW)*(value1+value2) 
              )

File: Arcade/3a-TheCore/IntroGates/knapsackLight.go
// https://app.codesignal.com/arcade/code-arcade/at-the-crossroads/r9azLYp2BDZPyzaG2
func knapsackLight(value1 int, weight1 int, value2 int, weight2 int, maxW int) int {
    
    // What a mess!
    if (weight1 + weight2) <= maxW {
        return value1 + value2;
    }
    if weight1 <= maxW {
        if value1 > value2 {
            return value1;
        } else if weight2 > maxW {
            return value1;
        }
    }
    if weight2 <= maxW {
        return value2;
    } else { 
        return 0; 
    }
       
}
